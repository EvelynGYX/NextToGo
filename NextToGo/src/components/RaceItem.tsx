import React from 'react';
import {StyleSheet, Text, View} from 'react-native';
import {ChevronRight} from '../assets/Chevron';
import RaceIcon from '../assets/RaceIcon';
import {Race} from '../interfaces/RaceArrayType';

interface RaceItemProps extends Race {}

const RaceItem = (props: RaceItemProps) => {
  const {meetingName, venueState, raceNumber, category, advertisedStart} =
    props;

  return (
    <View style={styles.container}>
      <View style={styles.leftContainer}>
        <RaceIcon category={category} style={styles.leftIcon} />
        <View style={styles.leftContent}>
          <Text>{`${meetingName} R${raceNumber}`}</Text>
          <Text>{venueState}</Text>
        </View>
      </View>
      <View style={styles.rightContent}>
        <Text>{advertisedStart}</Text>
        <View style={styles.rightIcon}>
          <ChevronRight />
        </View>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingHorizontal: 12,
    paddingVertical: 8,
  },
  leftContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
  },
  leftContent: {
    flexDirection: 'column',
  },
  rightContent: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
  },
  rightIcon: {
    paddingLeft: 4,
  },
  leftIcon: {
    paddingRight: 24,
  },
});

export default RaceItem;
