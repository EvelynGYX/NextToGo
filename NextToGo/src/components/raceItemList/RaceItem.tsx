import React from 'react';
import {StyleSheet, Text, View} from 'react-native';
import {IRace} from '../../interfaces/RaceArray';
import {ChevronRightIcon} from '../icons/ChevronIcon';
import RaceIcon from '../icons/RaceIcon';

export type RaceItemProps = IRace;

const RaceItem = (props: RaceItemProps) => {
  const {meetingName, venueState, raceNumber, category, advertisedStart} =
    props;

  return (
    <View style={styles.container}>
      <View style={styles.leftContainer}>
        <RaceIcon category={category} style={styles.leftIcon} />
        <View style={styles.leftContent}>
          <Text>{`${meetingName} R${raceNumber}`}</Text>
          <Text>{venueState}</Text>
        </View>
      </View>
      <View style={styles.rightContent}>
        <Text>{advertisedStart}</Text>
        <View style={styles.rightIcon}>
          <ChevronRightIcon />
        </View>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingHorizontal: 12,
    paddingVertical: 8,
  },
  leftContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
  },
  leftContent: {
    flexDirection: 'column',
  },
  rightContent: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
  },
  rightIcon: {
    paddingLeft: 4,
  },
  leftIcon: {
    paddingRight: 24,
  },
});

export default RaceItem;
