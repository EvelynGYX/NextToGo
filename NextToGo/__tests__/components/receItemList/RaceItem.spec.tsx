import React from 'react';
import renderer from 'react-test-renderer';
import RaceItem, {
  RaceItemProps,
} from '../../../src/components/raceItemList/RaceItem';
import Category from '../../../src/constants/Category';

describe('RaceItem', () => {
  it('renders correctly', () => {
    const mockProps: RaceItemProps = {
      raceId: 'some raceId',
      meetingName: 'some meetingName',
      raceNumber: 'some number' as any as number,
      category: 'some category' as any as Category,
      advertisedStart: 'some advertisedStart',
      venueState: 'some venueState',
    };
    const tree = renderer.create(<RaceItem {...mockProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
