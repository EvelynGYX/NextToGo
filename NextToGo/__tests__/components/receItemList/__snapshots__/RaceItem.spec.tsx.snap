// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RaceItem renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingHorizontal": 12,
      "paddingVertical": 8,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 20,
          },
          Object {
            "paddingRight": 24,
          },
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      󱗀
    </Text>
    <View
      style={
        Object {
          "flexDirection": "column",
        }
      }
    >
      <Text>
        some meetingName Rsome number
      </Text>
      <Text>
        some venueState
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text>
      some advertisedStart
    </Text>
    <View
      style={
        Object {
          "paddingLeft": 4,
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 18,
            },
            undefined,
            Object {
              "fontFamily": "Entypo",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;
